pico/types/src/main/java/io/helidon/pico/types/AnnotationAndValue.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.types;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents an annotation along with its value(s).
+ */
+public interface AnnotationAndValue {
+
+    /**
+     * The type name, e.g., {@link jakarta.inject.Named} -> "jakarta.inject.Named".
+     *
+     * @return the annotation type name
+     */
+    TypeName getTypeName();
+
+    /**
+     * The value.
+     *
+     * @return The string value of (all attributes of) the annotation (expressed as toString()), or null if no value is present
+     */
+    String getValue();
```

Comment:
Naming of methods is not according to Helidon code style. We use "record" style (valid for all methods)

Reply:
Right, these should be called `typeName()` and `value()` to be consistent with Helidon conventions.

Reply:
I still think this is the wrong decision, just for the record. These APIs were meant to mirror TypeName -> https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/Type.html and AnnotationAndValue -> https://docs.oracle.com/javase/8/docs/api/javax/lang/model/element/AnnotationValue.html

But it seems like consistency to the rest of Helidon for the API contracts is paramount, so I ended up pushing the suggested changes.